cmd/testscript/main.go

```diff
@@ -282,5 +290,16 @@ func run(runDir, filename string, verbose bool, envVars []string) error {
 		return fmt.Errorf("error running %v in %v\n", renderFilename(filename), runDir)
 	}
 
+	if update {
+		// Write the testscript archive back to the source file
+		source, err := ioutil.ReadFile(scriptFile)
```

Comment:
Hasn't this file got the Go proxy files stripped out of it, and so won't this be losing that info from the script instead of just changing the `cmp` results? I think you probably need a merge.

Reply:
Excellent point. Will include a test for this too, per your comment.